src/AnimatedSprite.cpp

```diff
@@ -81,24 +87,50 @@ sf::FloatRect AnimatedSprite::getGlobalBounds() const
 
 void AnimatedSprite::setHighlight(bool h)
 {
-    if (h)
-    {
-        _highlight.setPosition(getPosition());
-        auto b = getGlobalBounds();
-        _highlight.setSize(sf::Vector2f{ b.width, b.height });
-    }
-    else
+    _isHighlighted = h;
+    setHighlightPosition();
+}
+
+void AnimatedSprite::draw(  sf::RenderTarget& target, 
+                            sf::RenderStates states     ) const
+{
+
+    states.transform *= getTransform();
+    target.draw(_sprite, states);
+
+    if(_isHighlighted)
     {
-        _highlight.setSize(sf::Vector2f{ 0.f, 0.f });
+        target.draw(_highlight);
     }
 
 }
 
-void AnimatedSprite::draw(sf::RenderTarget & target, sf::RenderStates states) const
+void AnimatedSprite::setHighlightPosition()
 {
-    states.transform *= getTransform();
-    target.draw(_sprite, states);
-    target.draw(_highlight);
+    auto b = getGlobalBounds();
+    
+    float x = b.left + ((b.width / 2)   - (DEFAULT_HIGHLIGHT_WIDTH / 2));
+    float y = b.top + ((b.height)       - (DEFAULT_HIGHLIGHT_HEIGHT));
+    
+    _highlight.setPosition( sf::Vector2f{ x, y } );
+    
+    _highlight.setSize(sf::Vector2f{DEFAULT_HIGHLIGHT_WIDTH,
+                                    DEFAULT_HIGHLIGHT_HEIGHT } ); 
+
+    _highlight.setOutlineThickness(2);
+
+}
+
+void AnimatedSprite::setPosition(float x, float y)
```

Comment:
Neither overload of `sf::Transformable::setPosition()` is marked as virtual, so this is effectively hiding the call, which is bad. This why I set the position in `timestep()`.

Reply:
I notice that the coords are not correct on the highlight initially if it is not set in setPosition().

The coords become correct after calling setHighlight(). So this was breaking my bounds checking unless the user enabled debug mode first. Setting this in setPosition() fixes that issue.